tests/gtests/test_pooling_forward.cpp

```diff
@@ -785,12 +777,7 @@ INSTANTIATE_TEST_SUITE_P(TestPoolingForwardZeroDim, pooling_test_float,
                         algorithm::pooling_max, memory::format_tag::nhwc,
                         memory::format_tag::nhwc,
                         EXPAND_SIZES_2D(
-                                0, 4, 4, 4, 4, 4, 3, 3, 0, 0, 1, 1, 1, 1)},
-                pool_test_params_float {prop_kind::forward_training,
-                        algorithm::pooling_max, memory::format_tag::nchw,
-                        memory::format_tag::nchw,
-                        EXPAND_SIZES_2D(
-                                2, 4, 0, 4, 4, 4, 3, 3, 1, 1, 1, 1, 1, 1)}));
+                                0, 4, 4, 4, 4, 4, 3, 3, 0, 0, 1, 1, 1, 1)}));
```

Comment:
With the test updated, I feel most changes from the library can be dropped. Could you confirm, please?

Reply:
@dzarukin I think even though we removed the unsupported test and benchdnn doesn't allow cases with ih==0, it is still possible to run such a case if using the oneDNN API programmatically. Additionally, the cuda implementation has this behaviour implemented (https://github.com/oneapi-src/oneDNN/blob/main/src/gpu/nvidia/cudnn_pooling.cpp#L43-L45). Should I remove the library changes in this PR and add a check in the `init()` for input dimensions != 0 (except C dimension)? Or is this a check that should be added in the primitive desc create so it applies to all pooling implementations? Or does it make sense to keep the changes in this PR?

Reply:
Zero spatial has no sense for pooling operation, it shouldn't be used by users and must be prohibited on the library API side (so it's not an impl level, it's a higher level). I'll take care of it. Please remove all related to zero-spatial changes as they are not supposed to be needed. If there are any changes needed for zero-batch or zero-channel cases, those should remain. Thanks.

Reply:
Removed the library changes as they are not required in this case, thanks.